### PR TITLE
1.12.2

### PR DESCRIPTION
**Pull request by Wyn-Price**
It would be a good idea to also create a new branch for 1.12.2. Also if i could have access to this repo so I can make changes to the code without having to create a PR every time

### CODE REVIEW

**Comment by AbrarSyed** [comment]
I totally agree with you. After this initial PR to put your stuff in, il give you contributor access so you can manage it as you see fit.

Now.. I have some changes I would like to request.

1. Keep the old gitignore. I honestly think the old one is superior to the one you replaced it with. Maybe add the few lines to ignore the hsperrid and bluej files.
2. Keep the License file (assuming ofcrouse its still LGPL) or replace with a different appropriate license file.
3. Keep most of the old readme. Sure change the description up top, but the stuff like links to curseforge and forum posts as well as building instructions should stay. [Good info on readmes](https://github.com/LappleApple/feedmereadmes/blob/master/README-maturity-model.md).
4. Keep the dpeloyment part of the build.gradle. basically the entire `uploadArchives` part and related sections. That's important for the jenkins stuff.

Also see my other in-line comments

File: build.gradle [hunk 1]
```diff
@@ -1,168 +1,64 @@
...
 minecraft {
-    version = "1.7.10-10.13.2.1291" // grab latest forge
+    version = "1.12.2-14.23.0.2512"
     runDir = "run"
-
-    // replacing
-    replace "@CHANGE_VERSION@", "[4.7,)"
-    replace "@MC_VERSION@", project.minecraft.version
-    replace "@VERSION@", project.version
-    replaceIn "SecretRooms.java"
```
Comment (AbrarSyed) [inline build.gradle]: You should probably change these values, but not remove them entirely. I might have been out of the community for quite a while, but Im still pretty sure this info is useful to keep in the `@Mod`.

File: build.gradle [hunk 2]
```diff
@@ -1,168 +1,64 @@
...
+        classpath 'net.minecraftforge.gradle:ForgeGradle:2.3-SNAPSHOT'
     }
-    dependencies { classpath 'net.minecraftforge.gradle:ForgeGradle:1.2-SNAPSHOT' }
 }
+apply plugin: 'net.minecraftforge.gradle.forge'
+//Only edit below this line, the above code adds and enables the necessary things for Forge to be setup.
+
 
-apply plugin: "forge"
+version = "2.0.0"
```
Comment (AbrarSyed) [inline build.gradle]: should start form version `5.0.0` so it doesnt show up as an earlier version that the existing `4.7.1` and older.

File: src/main/java/com/github/abrarsyed/secretroomsmod/common/SecretRooms.java [hunk 3]
```diff
@@ -1,685 +0,0 @@
...
-import cpw.mods.fml.common.event.FMLInterModComms;
-import cpw.mods.fml.common.event.FMLPreInitializationEvent;
-import cpw.mods.fml.common.event.FMLServerStartingEvent;
-import cpw.mods.fml.common.event.FMLServerStoppingEvent;
-import cpw.mods.fml.common.registry.GameRegistry;
-
-/**
- * @author AbrarSyed
- */
-@Mod(modid = SecretRooms.MODID, version = "@VERSION@", useMetadata = true,
```
Comment (AbrarSyed) [inline src/main/java/com/github/abrarsyed/secretroomsmod/common/SecretRooms.java]: Notice the tokens here that are replaced by the buildscript

**Comment by Wyn-Price** [comment]
I had to change the .gitignore as it spammed an immense amount of eclipse files into git. I can work out what to change and add the lines

File: build.gradle [hunk 4]
```diff
@@ -62,3 +67,85 @@ processResources {
...
+}
+
+
+uploadArchives {
+    repositories {
+        mavenDeployer {
+            configuration = configurations.deployerJars
+
+            repository(url: "http://files.minecraftforge.net/maven/manage/upload") {
+                authentication(userName: "AbrarSyed", password: project.hasProperty(forgeMavenPass) ? null : project.getProperty('forgeMavenPass')) // the elvis operator. look it up.
```
Comment (Wyn-Price) [inline build.gradle]: @AbrarSyed I havnt set up an account or anything for this, and ill probally need some help setting all this up. I have made APIs like this in the past, but using my website as a maven. I dont currently have anything for an API to be used for but Im sure itll come it usfull later.

Comment (AbrarSyed) [reply build.gradle]: The password for that is injected at build-time by the forge jenkins. Once you get a set of your own credentials, we can change it.

Upon deployment, it magically shows up here: http://files.minecraftforge.net/SecretRoomsMod/
About that.. Im worried the group needs to be the same for it to work correctly.. il double check with Overmind on that... In maven the group decides the file structure, and that script is generated by crawling the file structure... so.. uh.. yeah..

Comment (Wyn-Price) [reply build.gradle]: So will Overmind set me up with an account or is that something I will have to do. Ive used maven before but I used it through GitHub and GitHub pages

Comment (AbrarSyed) [reply build.gradle]: Dunno yet, Il find out. For now just leave my credentials in there.

Comment (Wyn-Price) [reply build.gradle]: ok

**Comment by Wyn-Price** [comment]
Ok Im pretty much done with this for now. Unless theres somthing else i could add or there bugs that need squashing I think theres not much more I can do with this. If you want to merge the branches you can, or you can keep it how it is i dont mind

**Pull request merged**